git/week-1/lesson.md

```diff
@@ -0,0 +1,107 @@
+# Github and Github Desktop - Week 1
+
+## Learning Objectives
+
+- You will be able to create a repository on Github
+- You will be able to clone a repository using Github Desktop
+- You will be able to make changes to a repository using Github Desktop
+- You can define what the differences between Git and Github are
+- You can use git to submit all your homeworks to Github
+
+---
+
+## What is the problem?
+
+When programmers work together it can be hard to stay in sync. This can be demonstrated by:
+
+- Have one person write three words on a piece of paper
+- Give the paper to the teacher and the teacher makes a copy of it
+- Somebody else can request the paper from the teacher
+- Change a word on the paper and give it back to the teacher.
+
+The teacher is the single source of truth for the project. This is fine so long as both people don't make a change at the same time.
+
+- Now both students change a word on their piece of paper
+- They then try to give both pieces of paper to the teacher
+  - But the teacher doesn't know what to do! Which one is correct?
+
+This is why Git was invented!
+
+## What is Github?
+
+**Confirm that everyone has a Github account and that they've been added to the relevant Github Teams**
+
+1. Give an overview of Github including
+   - A Project Overview
+   - Issues
+   - Creating a repository
+   - Changing a file in a repository
+2. All students should now
+   - Create a repository called "Code Your Future". This should include the ReadMe.md.
+   - Edit the ReadMe.md inside Github to include their name.
+
+## What is Github Desktop?
+
+Github desktop is a way to edit files on your computer instead of editing them online.
+
+Give a demonstration of
+
+1. Cloning a repository
+2. Making a change to a file in the repository
+3. Committing the changes to our local repository
+4. Pushing our changes to Github
+
+The students should now repeat these steps on their projects
+
+## What about working in Teams?
```

Comment:
Do we actually get the students to work in teams doing their homework? I wonder if this should go later in the course? On the other hand, I do think this is an interesting exercise to show the difference between git and say Dropbox.

Also, minor side point: why is teams capitalised? Seems to imply that it's a product or something.